internal/provider/data_source_azure_peering_connection.go

```diff
@@ -0,0 +1,147 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceAzurePeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description: "The Azure peering connection data source provides information about a peering connection between an HVN and a peer Azure VNet.",
+		ReadContext: dataSourceAzurePeeringConnectionRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &peeringCreateTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"peering_id": {
+				Description: "The ID of the peering connection.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"hvn_link": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			// Optional inputs
+			"wait_for_active_state": {
+				Description: "If `true`, Terraform will wait for the peering connection to reach an `ACTIVE` state before continuing. Default `false`.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+			},
+			// Computed outputs
+			"organization_id": {
+				Description: "The ID of the HCP organization where the peering connection is located. Always matches the HVN's organization.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the HCP project where the peering connection is located. Always matches the HVN's project.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"application_id": {
+				Description: "The application ID of the HCP VNet backing the HVN.",
```

Comment:
I think the following text may make more sense: "The application id is the ID of Azure application whose credentials are used to peer HCP HVN VNet with customer VNet."